llvm/lib/Transforms/Utils/SCCPSolver.cpp

```diff
@@ -788,6 +794,17 @@ class SCCPInstVisitor : public InstVisitor<SCCPInstVisitor> {
       markOverdefined(ValueState[V], V);
   }
 
+  void trackValueOfArgument(Argument *A) {
+    if (A->getType()->isIntegerTy()) {
```

Comment:
Do we need the restriction to non-vectors here? IIRC ipsccp can handle vectors though it mostly doesn't.

Reply:
I only assumed that vectors was not handled as there was a restriction to only non-vectors for the range metadata.

Reply:
Ok, we can relax this separately from this PR.

Reply:
Tried to add a test for vector like below but was not optimized when I remove the check. But maybe I did something wrong.
```
define void @range_attribute_vec(<2 x i32> range(i32 0, 10) %v) {
  %c1 = icmp ult <2 x i32> %v, <i32 10, i32 10>
  call void @use(<2 x i1> %c1)
  %c2 = icmp ult <2 x i32> %v, <i32 9, i32 9>
  call void @use(<2 x i1> %c2)
  %c3 = icmp ugt <2 x i32> %v, <i32 9, i32 9>
  call void @use(<2 x i1> %c3)
  %c4 = icmp ugt <2 x i32> %v, <i32 8, i32 8>
  call void @use(<2 x i1> %c4)
  ret void
}
```

Reply:
think that my test fails as ConstantVector is not handled in https://github.com/andjo403/llvm-project/blob/fa71860e1937b916758693e268d5a16021c8b5c3/llvm/include/llvm/Analysis/ValueLattice.h#L301-L339